src/machine/machine-dbus.c

```diff
@@ -638,6 +640,55 @@ int bus_machine_method_open_shell(sd_bus_message *message, void *userdata, sd_bu
         if (!strv_env_is_valid(env))
                 return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid environment assignments");
 
+        /* if we are a VM we need to transform args/path/env into an SSH command line */
+        if (m->class == MACHINE_VM) {
```

Comment:
hmm, so you are running ssh server side?

i think it would be less surprising if we did this client side. i.e. in "machinectl shell" we try to use OpenShell() first. Which will fail on a VM, and then we query the SSH details instead, and chainload ssh instead?

Reply:
So I've moved this to machinectl now:
- added GetSSHInfo to the Machine object and the properties, VsockCid, SshAddress, SshPrivateKeyPath.
- added GetMachineSSHInfo to the Manager object to provide a convenience method.
- moved SSH cmdline building logic into machinectl
- dropped previous OpenMachineShell commit

I still have the old branch locally so can restore if we want to go back to that way of doing things.